### ISSUE TITLE
FTL fails on default Raspberry Pi install.

### ISSUE THREAD

**Issue by diksm8** [pi-hole/pi-hole#2992]
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*.
- [X] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

---
During install FTL fails and internet stops working. Default buster install, just flashed image and ran the script.

```
pi@raspberrypi:~/Pi-hole/automated install $ sudo bash basic-install.sh

  [✓] Root user check

        .;;,.
        .ccccc:,.
         :cccclll:.      ..,,
          :ccccclll.   ;ooodc
           'ccll:;ll .oooodc
             .;cll.;;looo:.
                 .. ','.
                .',,,,,,'.
              .',,,,,,,,,,.
            .',,,,,,,,,,,,....
          ....''',,,,,,,'.......
        .........  ....  .........
        ..........      ..........
        ..........      ..........
        .........  ....  .........
          ........,,,,,,,'......
            ....',,,,,,,,,,,,.
               .',,,,,,,,,'.
                .',,,,,,'.
                  ..'''.

  [✓] Disk space check
  [✓] Update local cache of available packages

  [✓] Checking apt-get for upgraded packages... up to date!

  [i] Installer Dependency checks...
  [✓] Checking for apt-utils
  [i] Checking for dialog (will be installed)
  [✓] Checking for debconf
  [✓] Checking for dhcpcd5
  [✓] Checking for git
  [✓] Checking for iproute2
  [✓] Checking for whiptail
  [i] Using interface: eth0
  [i] Using Google (ECS)
  [✓] Set IP address to 192.168.1.118 
  You may need to restart after the install is complete
  [i] Unable to find IPv6 ULA/GUA address, IPv6 adblocking will not be enabled
  [i] IPv4 address: 192.168.1.118/24
  [i] IPv6 address: 
  [i] Web Interface On
  [i] Web Server On
  [i] Logging On.
  [✗] Check for existing repository in /etc/.pihole
  [✓] Clone https://github.com/pi-hole/pi-hole.git into /etc/.pihole

  [✗] Check for existing repository in /var/www/html/admin
  [✓] Clone https://github.com/pi-hole/AdminLTE.git into /var/www/html/admin

  [i] Main Dependency checks...
  [✓] Checking for cron
  [✓] Checking for curl
  [i] Checking for dnsutils (will be installed)
  [✓] Checking for iputils-ping
  [i] Checking for lsof (will be installed)
  [i] Checking for netcat (will be installed)
  [✓] Checking for psmisc
  [✓] Checking for sudo
  [✓] Checking for unzip
  [✓] Checking for wget
  [i] Checking for idn2 (will be installed)
  [i] Checking for sqlite3 (will be installed)
  [✓] Checking for libcap2-bin
  [i] Checking for dns-root-data (will be installed)
  [i] Checking for resolvconf (will be installed)
  [✓] Checking for libcap2
  [i] Checking for lighttpd (will be installed)
  [i] Checking for php-common (will be installed)
  [i] Checking for php-cgi (will be installed)
  [i] Checking for php-sqlite3 (will be installed)
  [✓] Enabling lighttpd service to start on reboot...

  [i] FTL Checks...

  [✓] Detected ARM-hf architecture (armv7+)
  [i] Checking for existing FTL binary...
  [✗] Downloading and Installing FTL
  Error: Unable to get latest release location from GitHub
  [✗] FTL Engine not installed
```
```
pi@raspberrypi:~ $ ping 8.8.8.8
PING 8.8.8.8 (8.8.8.8) 56(84) bytes of data.
64 bytes from 8.8.8.8: icmp_seq=1 ttl=56 time=6.30 ms
64 bytes from 8.8.8.8: icmp_seq=2 ttl=56 time=6.64 ms
^C
--- 8.8.8.8 ping statistics ---
2 packets transmitted, 2 received, 0% packet loss, time 2ms
rtt min/avg/max/mdev = 6.299/6.469/6.639/0.170 ms
pi@raspberrypi:~ $ nslookup google.com
;; connection timed out; no servers could be reached
```

**Comment by dschaper** [comment]
Change `/etc/resolv.conf` from `127.0.0.1` to something external like `8.8.8.8` or `1.1.1.1`

**Comment by diksm8** [comment]
How do I complete the install after changing the resolv config?

**Comment by dschaper** [comment]
Run the install again.

**Comment by Powderwt** [comment]
i too am having issues here, however i do not know how to Change /etc/resolv.conf from '127.0.0.1' to something external like '8.8.8.8' 

Also, when I try 'sudo service dnsmasq status' i get 'Unit dnsmasq.service could not be found.'

'$ dig google.com @8.8.8.8

; <<>> DiG 9.11.5-P4-5.1-Raspbian <<>> google.com @8.8.8.8
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 19030
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 512
;; QUESTION SECTION:
;google.com.                    IN      A

;; ANSWER SECTION:
google.com.             299     IN      A       172.217.4.78

;; Query time: 25 msec
;; SERVER: 8.8.8.8#53(8.8.8.8)
;; WHEN: Sun Nov 10 18:13:33 EST 2019
;; MSG SIZE  rcvd: 55'

**Comment by dschaper** [comment]
`sudo nano /etc/resolv.conf` and tehre is no dnsmasq anymore, it's `pihole-FTL`.

**Comment by afferrell** [comment]
I have the same problem.
`/etc/resolv.conf` doesn't contain any ip addresses when I open it with Vim. However, when I view the file on my computer it's listed as a broken link. The target is `/etc/resolvconf/run/resolv.conf.` The file Vim is opening is `/etc/resolvconf/resolv.conf.d/original`.

**Comment by Powderwt** [comment]
> I have the same problem.
> `/etc/resolv.conf` doesn't contain any ip addresses when I open it with Vim. However, when I view the file on my computer it's listed as a broken link. The target is `/etc/resolvconf/run/resolv.conf.` The file Vim is opening is `/etc/resolvconf/resolv.conf.d/original`.
> 
> Edit: Right next to `resolv.conf` was `resolv.conf.bak` _that_ file seemed more like what you described. So, I deleted `resolv.conf` and renamed the other file to replace it. Now everything seems to be working again.

I was able to fix mine by replacing the IP in the resolve.conf file `127.0.0.1` with `9.9.9.9`

I've had everything up and running for a while now.

Has really slowed down my DL rates with some unraid Dockers however, but that's a problem for another forum for another day I reckon.

**Comment by moranmoa** [comment]
I'm having the same problem 

curl -sSL https://install.pi-hole.net | bash

  [✗] Root user check
  [i] Script called with non-root privileges
      The Pi-hole requires elevated privileges to install and run
      Please check the installer for any concerns regarding this requirement
      Make sure to download this script from a trusted source

  [✓] Sudo utility check

  [✓] Root user check

        .;;,.
        .ccccc:,.
         :cccclll:.      ..,,
          :ccccclll.   ;ooodc
           'ccll:;ll .oooodc
             .;cll.;;looo:.
                 .. ','.
                .',,,,,,'.
              .',,,,,,,,,,.
            .',,,,,,,,,,,,....
          ....''',,,,,,,'.......
        .........  ....  .........
        ..........      ..........
        ..........      ..........
        .........  ....  .........
          ........,,,,,,,'......
            ....',,,,,,,,,,,,.
               .',,,,,,,,,'.
                .',,,,,,'.
                  ..'''.

  [✓] Disk space check
  [✓] Update local cache of available packages

  [✓] Checking apt-get for upgraded packages... 11 updates available
  [i] It is recommended to update your OS after installing the Pi-hole!

  [i] Installer Dependency checks...
  [✓] Checking for apt-utils
  [i] Checking for dialog (will be installed)
  [✓] Checking for debconf
  [✓] Checking for dhcpcd5
  [i] Checking for git (will be installed)
  [✓] Checking for iproute2
  [✓] Checking for whiptail
  [i] Using Google (ECS)
  [✓] Set IP address to 192.168.1.115
  You may need to restart after the install is complete
  [i] Unable to find IPv6 ULA/GUA address, IPv6 adblocking will not be enabled
  [i] IPv4 address: 192.168.1.115/24
  [i] IPv6 address:
  [i] Web Interface On
  [i] Web Server On
  [i] Logging On.
  [✗] Check for existing repository in /etc/.pihole
  [✓] Clone https://github.com/pi-hole/pi-hole.git into /etc/.pihole

  [✗] Check for existing repository in /var/www/html/admin
  [✓] Clone https://github.com/pi-hole/AdminLTE.git into /var/www/html/admin

  [i] Main Dependency checks...
  [✓] Checking for cron
  [✓] Checking for curl
  [i] Checking for dnsutils (will be installed)
  [✓] Checking for iputils-ping
  [i] Checking for lsof (will be installed)
  [i] Checking for netcat (will be installed)
  [✓] Checking for psmisc
  [✓] Checking for sudo
  [✓] Checking for unzip
  [✓] Checking for wget
  [i] Checking for idn2 (will be installed)
  [i] Checking for sqlite3 (will be installed)
  [✓] Checking for libcap2-bin
  [i] Checking for dns-root-data (will be installed)
  [i] Checking for resolvconf (will be installed)
  [✓] Checking for libcap2
  [i] Checking for lighttpd (will be installed)
  [i] Checking for php-common (will be installed)
  [i] Checking for php-cgi (will be installed)
  [i] Checking for php-sqlite3 (will be installed)
  [✓] Enabling lighttpd service to start on reboot...
  [✓] Creating user 'pihole'

  [i] FTL Checks...

  [✓] Detected ARM-hf architecture (armv6 or lower) Using ARM binary
  [i] Checking for existing FTL binary...
  [i] Downloading and Installing FTL...curl: (6) Could not resolve host: github.com
  [✗] Downloading and Installing FTL
  Error: URL https://github.com/pi-hole/ftl/releases/latest/download/pihole-FTL-arm-linux-gnueabi not found
  [✗] FTL Engine not installed


and my /etc/resolv.conf don't have any IP only this:

# Generated by dhcpcd
# /etc/resolv.conf.head can replace this line
# /etc/resolv.conf.tail can replace this line

**Comment by moranmoa** [comment]
anyone have an older version install witch works?

**Comment by m-thomson** [comment]
Same problem here. FWIW, adding `8.8.8.8` to `resolv.conf` after rebooting fixed the problem and allowed the install to complete.

**Comment by angrynerd** [comment]
Afaik if you are still having this issue after changing `resolv.conf` then you can either A. Manually install dnsmasq which is what i did the first time. Pi hole removes it during the script so the manual install doesn't interfere but it means that all of the conf files for it already exist. B. Try and do the same thing as listed in the solution to using this with pi hole and create the leases file yourself and give pi hole permissions to it. C. Run the script as root by typing `curl -sSL https://install.pi-hole.net | sudo bash`

**Comment by dschaper** [comment]
We don't remove `dnsmasq`. We disable the daemon startup if the service is installed but we don't actually remove it.

**Comment by angrynerd** [comment]
My apologies i may have misread during my install as i though it said uninstalling dnsmasq, but it would make sense if it didn't. Either way installing it prior got me past that issue.

**Comment by MikePadge** [comment]
As others have reported this being a DNS issue, I'm seeing SSL cert issues with the curl command on raspbian 10. 

I haven't dug into the root certs that are possibly missing, but this looks to be the crux of the problem. 

TLDR HACK FIX: Not recommended due to the obvious security issues, but adding the ignore SSL -k flag to the curl commands in the basic-install.sh script hackily gets this rolling. 

It looks like Raspbian 10 / Curl certs might be missing a major root cert? Probably something Comodo/Sectigo related if I had to guess.

**Comment by dschaper** [comment]
Don't ignore the SSL please.

I was updating the infrastructure, you may have just hit things when I was restarting the proxy.
